tests/test_healpix_padding.py

```diff
@@ -0,0 +1,94 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import matplotlib.pyplot as plt
+import numpy
+import pytest
+import torch
+
+from earth2grid.healpix import XY, Grid, PaddingBackends, pad, pad_backend, pad_with_dim
+
+
+def test_hpx_pad(regtest):
+    order = 2
+    nside = 2**order
+    face = 5
+
+    grid = Grid(order, pixel_order=XY())
+    lat = torch.from_numpy(grid.lat)
+    lon = torch.from_numpy(grid.lon)
+    z = lon + 3 * lat
+    padded = pad_with_dim(z, padding=nside, dim=-1)
+    m = nside + 2 * nside
+    padded = padded.reshape(12, m, m)
+
+    for face in range(padded.shape[0]):
+        print(f"{face=}", file=regtest)
+        numpy.savetxt(regtest, padded[face].cpu(), fmt="%.2f")
+
+
+def test_hpx_pad_versus_zephyr(tmp_path):
+    order = 3
+    nside = 2**order
+
+    grid = Grid(order, pixel_order=XY())
+    lat = torch.from_numpy(grid.lat)
+    lon = torch.from_numpy(grid.lon)
+    z = lon + 3 * lat
+    z = torch.arange(grid.shape[0])
+
+    z = z.reshape(1, 12, nside, nside).float()
+
+    with pad_backend(PaddingBackends.zephyr):
+        expected = pad(z, padding=nside)
+
+    with pad_backend(PaddingBackends.indexing):
```

Comment:
Would it also make sense to add the cuda backend to this test?

Reply:
that wouldn't be able to run on github actions, so I didn't bother. in the medium term, after testing this code, i would like to remove both the cuda and zephyr implementation. The regression tests should let us do that.